go/cmd/internal/docgen/docgen_test.go

```diff
@@ -0,0 +1,134 @@
+/*
+Copyright 2024 The Vitess Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package docgen
+
+import (
+	"testing"
+
+	"github.com/spf13/cobra"
+	"github.com/stretchr/testify/require"
+)
+
+func TestGenerateMarkdownTree(t *testing.T) {
+	tests := []struct {
+		name        string
+		dir         string
+		cmd         *cobra.Command
+		expectedErr string
+	}{
+		{
+			name:        "Empty dir",
+			dir:         "",
+			cmd:         &cobra.Command{},
+			expectedErr: "mkdir : no such file or directory",
```

Comment:
imo we should not assert on the literal text of error messages. it's brittle and our tests can break simply due to golang upgrades, library updates, etc (since we don't define these messages ourselves)

Reply:
I don't understand the value of asserting on the error message at all really, as opposed to asserting "we got an error"